File: wanandroid/src/main/java/com/oreooo/wanandroid/search/SearchContract.java
package com.oreooo.wanandroid.search;

import com.oreooo.baselibrary.mvp.BaseContract;
import com.oreooo.baselibrary.pojo.Article;

public interface SearchContract {
    interface Presenter extends BaseContract.BasePresenter {
        void getSearchArticle(int page, String keyword);
    }

    interface View extends BaseContract.BaseView {
        void showSearchResult(Article data);
    }
}

File: module_search/src/main/java/com/oreo/module_search/SearchContract.kt
package com.oreo.module_search

import com.oreooo.baselibrary.mvpbase.BaseContract
import com.oreooo.baselibrary.pojo.Article

interface SearchContract {
    interface View : BaseContract.BaseView {
        fun showArticleResult(data: Article)
    }

    interface Presenter : BaseContract.BasePresenter {
        fun getSearchArticle(page: Int, keyword: String)
    }
}